File: core/src/main/java/xyz/qalcyo/requisite/core/cosmetics/ICosmeticInitializer.java
package xyz.qalcyo.requisite.core.cosmetics;

import java.util.List;

public interface ICosmeticInitializer<T> {
    void initialize(List<ICosmetic<T>> cosmetics, CosmeticData data);
    void finish(CosmeticManager<T> cosmeticManager, List<ICosmetic<T>> cosmetics);
    ICosmeticRenderer<T> renderer();
}

File: core/src/main/kotlin/xyz/qalcyo/requisite/core/cosmetics/ICosmeticInitializer.kt
package xyz.qalcyo.requisite.core.cosmetics

interface ICosmeticInitializer<T> {
    fun initialize(cosmetics: List<ICosmetic<T>?>?, data: CosmeticData?)
    fun finish(cosmeticManager: CosmeticManager<T>?, cosmetics: List<ICosmetic<T>?>?)
    fun renderer(): ICosmeticRenderer<T>?
}